### PR TITLE
[build] remove the use of NO_CUDA

### PR DESCRIPTION
**Pull request by Yangqing**
I wanted to do a check to see how difficult it is to remove deprecated flags. This one deals with NO_CUDA.

### CODE REVIEW

**Comment by apaszke** [comment]
Considering that all other flags in `setup.py` (that even appear right next to this one) are:

* NO_CUDNN
* NO_MKLDNN
* NO_NNPACK
* NO_DISTRIBUTED
* NO_SYSTEM_NCCL

Why would we want to change this one to USE_CUDA?

**Comment by Yangqing** [comment]
Plan on record is to move all of them to USE_ at least on the cmake side, cc @orionr.

@soumith, would setup.py want to keep NO_* syntax or USE_* syntax? I am ok with both, and essentially someone needs to set up the bridges.

**Comment by soumith** [comment]
we will enforce the NO_* via setup.py if we decide to keep them. Feel free to simplify the cmake, seems like a good direction,

File: .jenkins/pytorch/build-asan.sh [hunk 1]
```diff
@@ -17,5 +17,5 @@ export ASAN_OPTIONS=detect_leaks=0:symbolize=1
 # TODO: Make the ASAN flags a more unified env var
 CC="clang" CXX="clang++" LDSHARED="clang --shared" \
   CFLAGS="-fsanitize=address -shared-libasan" \
-  NO_CUDA=1 DEBUG=1 \
+  USE_CUDA=0 DEBUG=1 \
```
Comment (ezyang) [inline .jenkins/pytorch/build-asan.sh]: Yeah, I just want to point out that passing `NO_CUDA=1` as an env var to the setup.py script is considered part of the externally facing interface, so you'll probably want to keep at least one of these to make sure the BC-layer (which will check `NO_CUDA` envvar if `USE_CUDA` is not set) is working.

**Comment by Yangqing** [comment]
sg, new PR now only changes the native build front and leaves setup.py to use NO_* syntax.

**Comment by Yangqing** [comment]
Build passing - @soumith @ezyang mind taking a look?

**Review by ezyang — DISMISSED** [review]
See comment

File: tools/setup_helpers/cuda.py [hunk 2]
```diff
@@ -58,8 +58,10 @@ def find_cuda_version(cuda_home):
     if len(candidates) > 0:
         return candidates[0]
 
-
-if check_env_flag('NO_CUDA') or check_env_flag('WITH_ROCM'):
+# Note(jiayq): when migrating from NO_CUDA to USE_CUDA, the former case is checking
+# with default False, so we need to preserve old behavior by checking USE_CUDA with
+# default ON.
+if not check_env_flag('USE_CUDA', 'ON') or check_env_flag('WITH_ROCM'):
```
[the remark this answers is not in the document]
Comment (Yangqing) [reply tools/setup_helpers/cuda.py]: Ah good point, it's a missing git reset command, let me update.

[the remark this answers is not in the document]
Comment (Yangqing) [reply tools/setup_helpers/cuda.py]: done, @ezyang  PTAL?

**Review by ezyang — APPROVED** [review]

**Pull request merged**